### PR TITLE
Add bash completions

### LINKED ISSUE #43: Add Bash completions
### Description of feature

Self explanatory, completions are nice :)

### Installed makedeb version

8.0.4-1

### PR DESCRIPTION
**Pull request by Elsie19**
When you run makedeb as root, it fails, so simply creating a user will make makedeb work in Docker

Also the indentations are messed up on the completions, idk why

Fixes #43

### CODE REVIEW

**Comment by lpuv** [comment]
`sudo` isn't a user iirc, it's a group? :P

**Comment by Elsie19** [comment]
Yup, it's a group. It's wheel on arch systems

**Comment by lpuv** [comment]
can you use adduser to add a group?

**Comment by Elsie19** [comment]
not that I know of

File: docker/Dockerfile [hunk 1]
```diff
@@ -15,4 +15,10 @@ RUN echo 'deb [signed-by=/usr/share/keyrings/makedeb-archive-keyring.gpg arch=al
 RUN apt-get update
 RUN apt-get install "${package}" -y
 
+RUN adduser --disabled-password --gecos '' makedeb
+RUN adduser makedeb sudo
```
Comment (lpuv) [inline docker/Dockerfile]: uh... this is what I meant

**Comment by hwittenborn** [comment]
Docker images are actually currently handled in the makedeb/makedeb-docker repository :P. That's my bad for not removing the docker/ folder, I'll try to get that done soon.

If you head over to that repository, the makedeb user is also already implemented for everything, so that should all be covered unless there was something else you were looking at adding.

The Bash completions look promising though! I'll give those a spin and let you know how that part goes. For the time being you can go ahead and just delete the docker/ folder too, as there's no plan to keep it.

**Review by hwittenborn — CHANGES_REQUESTED** [review]
Could you remove the `src/dependency_deb/` files? I don't see any reason why they would be needed.

File: src/completions/bash [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		-A|--ignore-arch)
+			return
+			;;
+
+		-d|--no-deps)
+			return
+			;;
+
+		-F|--file)
+			COMPREPLY=( $( compgen -W "$(\ls -1aA .)" -- $cur ) )
```
Comment (hwittenborn) [inline src/completions/bash]: This doesn't appear to be working on files that contain spaces (i.e. `hello world.txt`). Maybe something like this?

```sh
mapfile -t COMPREPLY < <(ls -1aA | grep "^${cur}" 2> /dev/null)
```

This has the downside that the user can't specify files that contain any regex, but I think this method would cover more common file names.

File: src/completions/bash [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+	case "$cur" in
+		-*)
+			COMPREPLY=( $( compgen -W '--ignore-arch --no-deps --file --gen-integ --help --help --field --install --version --rm-deps --sync-deps --print-control --print-srcinfo --skip-pgp-check --verbose --as-deps --no-confirm' -- "$cur" ) )
+			;;
+
+		*)
+			COMPREPLY=( $( compgen -W '-A -d -F -g -h -H -i -V -r -s' -- "$cur" ) )
+			;;
+	esac
```
Comment (hwittenborn) [inline src/completions/bash]: Why not have this at the bottom of the previous `case` statement?

File: src/completions/bash [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+} &&
```
Comment (hwittenborn) [inline src/completions/bash]: Why are we using `&&` (or anything in general) here?

File: src/completions/bash [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	esac
+
+	case "$cur" in
+		-*)
+			COMPREPLY=( $( compgen -W '--ignore-arch --no-deps --file --gen-integ --help --help --field --install --version --rm-deps --sync-deps --print-control --print-srcinfo --skip-pgp-check --verbose --as-deps --no-confirm' -- "$cur" ) )
+			;;
+
+		*)
+			COMPREPLY=( $( compgen -W '-A -d -F -g -h -H -i -V -r -s' -- "$cur" ) )
+			;;
```
Comment (hwittenborn) [inline src/completions/bash]: I'd prefer these be listed as just arrays:

```sh

		-*)
			COMPREPLY=('--ignore-arch' '--no-deps' '--file' '--gen-integ'
                                   '--help' '--field' '--install' '--version'
                                   '--rm-deps' '--sync-deps' '--print-control' '--print-srcinfo'
                                   '--skip-pgp-check' '--verbose' '--as-deps' '--no-confirm')
			;;

		*)
			COMPREPLY=('-A' '-d' '-F' '-g' '-h' '-H' '-i' '-V' '-r' '-s')
			;;
```

`-i`, `--install`, and `--as-deps` are also not available on the Arch Linux releases, though that'll require some changes in the Makefile that I can just deal with after merging.

**Review by hwittenborn — DISMISSED** [review]

File: src/completions/bash (hunk 2 above)
Comment (Elsie19) [reply src/completions/bash]: Well, the problem with mapfile is that it doesn't exist in zsh, which is useful becaues zsh can emulate bash completions, with the exception of mapfile

**Review by hwittenborn — APPROVED** [review]
Woop woop woop

**Review by hwittenborn — APPROVED** [review]
Yep yep yep yep yep

**Pull request merged**